### PR TITLE
Migrate from ava to jest, add lint checks, migrate options to new file

### PR DESCRIPTION
**Pull request by szeck87**
* Migrates from ava to jest
* Adds lint checks for src and test directories
* Moves options to a separate file

Note:  This does not map languages and options yet.

@prettydiff do you have documentation or a file somewhere in your repo that specifies which options work with specific languages?  For example, `ternaryline` is used with JavaScript and TypeScript, but not with CSS.  Is that explicitly defined or can be inferred in one of your files?

### CODE REVIEW

**Comment by szeck87** [comment]
@Glavin001 I followed your suggestion from a couple of days ago on setting up the options.  It doesn't like having a [`commonOptions`](https://github.com/Unibeautify/beautifier-prettydiff/blob/jest-options-migrate/src/options.ts#L2-L87) object and applying that spread to the JavaScript object in options.  Suggestions?

File: src/index.ts [hunk 1]
```diff
@@ -35,7 +32,7 @@ export const beautifier: Beautifier = {
     Riot: options.Riot,
     XTemplate: options.XTemplate
   },
-  beautify(data: BeautifierBeautifyData) {
+  beautify(data : BeautifierBeautifyData) {
```
Comment (Glavin001) [inline src/index.ts]: This is strange.

File: src/index.ts [hunk 2]
```diff
@@ -115,8 +112,7 @@ export const beautifier: Beautifier = {
       });
       const result = prettydiff(args);
       return resolve(result);
-    }) as any;
+    })as any;
```
Comment (Glavin001) [inline src/index.ts]: This is also strange.

File: src/options.ts [hunk 3]
```diff
@@ -1,9 +1,10 @@
-import { BeautifierOptions, OptionValues } from "unibeautify";
-
+import {BeautifierOptions, OptionValues} from "unibeautify";
```
Comment (Glavin001) [inline src/options.ts]: I prefer to have spaces in here.
https://unibeautify.com/docs/option-object-curly-spacing.html#javascript

File: src/index.ts (hunk 2 above)
Comment (szeck87) [reply src/index.ts]: CI Unibeautify did that.

File: src/index.ts (hunk 1 above)
Comment (szeck87) [reply src/index.ts]: CI Unibeautify did that.

File: src/options.ts [hunk 4]
```diff
@@ -23,7 +26,7 @@ const commonOptions = {
   ],
   preserve: [
     ["preserve_newlines"],
-    (options: OptionValues): string => {
+    (options : OptionValues): string => {
```
Comment (Glavin001) [inline src/options.ts]: This is weird.

File: src/index.ts [hunk 5]
```diff
@@ -1,124 +1,38 @@
...
+import { Beautifier, Language, BeautifierBeautifyData } from "unibeautify";
 const prettydiff = require("prettydiff2");
-
+import options from "./options";
 const pkg = require("../package.json");
-
 export const beautifier: Beautifier = {
   name: "Pretty Diff",
   package: pkg,
   options: {
```
Comment (Glavin001) [inline src/index.ts]: What about passing the imported `options` variable into this `options` property?

File: src/options.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+    },
+  ],
+  methodchain: [
+    ["break_chained_methods"],
+    (options: OptionValues): boolean => {
+      if (options.break_chained_methods === true) {
+        return false;
+      } else {
+        return true;
+      }
```
Comment (Glavin001) [inline src/options.ts]: Could simplify to:

```
return !(options.break_chained_methods === true);
```

File: src/options.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+  },
+  Visualforce: {
+    ...commonOptions,
+  },
+  XML: {
+    ...commonOptions,
+  },
+  XTemplate: {
+    ...commonOptions,
+  },
```
Comment (Glavin001) [inline src/options.ts]: If ALL of these are the same, feel free to use shortcuts:

```javascript
const supportedLanguages: string[] = ["C#", "ColdFusion", ...];
const options = supportedLanguages.reduce((options, language) => ({
  ...options,
  [language]: commonOptions
});
```

Comment (szeck87) [reply src/options.ts]: They won't be the same in the end.  I'm just trying to structure everything properly then start adding more options and adding them to the proper languages.

File: src/options.ts [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+  comments: [
+    ["indent_comments"],
+    (options: OptionValues): string => {
+      if (options.indent_comments === false) {
+        return "noindent";
+      } else {
+        return "indent";
+      }
+    },
+  ],
```
Comment (Glavin001) [inline src/options.ts]:
```
src/options.ts(86,7): error TS2322: Type '{ "C#": { comments: (string[] | ((options: OptionValues) => string))[]; cssinsertlines: string; e...' is not assignable to type 'BeautifierOptions'.
  Types of property '"C#"' are incompatible.
    Type '{ comments: (string[] | ((options: OptionValues) => string))[]; cssinsertlines: string; endcomma:...' is not assignable to type 'boolean | BeautifierLanguageOptionComplex | undefined'.
      Type '{ comments: (string[] | ((options: OptionValues) => string))[]; cssinsertlines: string; endcomma:...' is not assignable to type 'BeautifierLanguageOptionComplex'.
        Property 'comments' is incompatible with index signature.
          Type '(string[] | ((options: OptionValues) => string))[]' is not assignable to type 'boolean | "align_assignments" | "arrow_parens" | "brace_style" | "break_chained_methods" | "comma...'.
            Type '(string[] | ((options: OptionValues) => string))[]' is not assignable to type '[BeautifierOptionName[], BeautifierOptionTransformFunction]'.
              Property '0' is missing in type '(string[] | ((options: OptionValues) => string))[]'.
```

Comment (Glavin001) [reply src/options.ts]: Looks similar to https://github.com/Unibeautify/unibeautify/blob/master/test/beautifier/options.spec.ts#L277

File: src/options.ts [hunk 9]
```diff
@@ -0,0 +1,161 @@
+import { BeautifierOptions, OptionValues } from "unibeautify";
+const commonOptions = {
```
Comment (Glavin001) [inline src/options.ts]: Add `BeautifierOptions`

```
const commonOptions: BeautifierOptions = {
```

Comment (Glavin001) [reply src/options.ts]: Nope. Should be `BeautifierLanguageOptions`.

Comment (Glavin001) [reply src/options.ts]:
```
--- a/src/options.ts
+++ b/src/options.ts
@@ -1,5 +1,5 @@
-import { BeautifierOptions, OptionValues } from "unibeautify";
-const commonOptions = {
+import { BeautifierOptions, OptionValues, BeautifierLanguageOptions } from "unibeautify";
+const commonOptions: BeautifierLanguageOptions = {
   comments: [
     ["indent_comments"],
     (options: OptionValues): string => {
```

**Pull request merged**